octoprint_authldap/__init__.py

```diff
@@ -0,0 +1,271 @@
+			ldap_verifypeer = settings().get(
+				["plugins", "authldap", "ldap_tls_reqcert"])
+			verifypeer = ldap.OPT_X_TLS_HARD
+			if ldap_verifypeer == 'NEVER':
+				verifypeer = ldap.OPT_X_TLS_NEVER
+			elif ldap_verifypeer == 'ALLOW':
+				verifypeer = ldap.OPT_X_TLS_ALLOW
+			elif ldap_verifypeer == 'TRY':
+				verifypeer = ldap.OPT_X_TLS_TRY
+			elif ldap_verifypeer == 'DEMAND':
+				verifypeer = ldap.OPT_X_TLS_DEMAND
+			# elif ldap_verifypeer == 'HARD':
+			#   verifypeer = ldap.OPT_X_TLS_HARD
+			connection.set_option(ldap.OPT_X_TLS_REQUIRE_CERT, verifypeer)
+			try:
+				connection.start_tls_s()
+				self._logger.error("TLS connection established.")
+			except:
+				self._logger.error("Error initializing tls connection")
+				pass
+		self._logger.error("Finished creating connection")
+		return connection
+
+	def escapeLDAP(self, str):
+		reservedStrings = ['+', '=', '\\', '\r',
+						   '\n', '#', ',', '>', '<', '"', ';']
+		for ch in reservedStrings:
+			if ch in str:
+				str = str.replace(ch, '\\' + ch)
+		return str
+
+	# Softwareupdate hook
+
+	def get_update_information(self):
+		return dict(
+			authldap=dict(
+				displayName="AuthLDAP",
+				displayVersion=self._plugin_version,
+
+				# version check: github repository
+				type="github_release",
+				user="gillg",
+				repo="OctoPrint-LDAP",
+				current=self._plugin_version,
+
+				# update method: pip
+				pip=("https://github.com"
+					 "/gillg/OctoPrint-LDAP/archive/{target_version}.zip")
+			)
+		)
+
+	# UserManager hook
+
+	def ldap_user_factory(components, settings, *args, **kwargs):
+		return LDAPUserManager()
+
+	# SettingsPlugin
+
+	def get_settings_defaults(self):
+		return dict(
+			active=False,
+			ldap_uri=None,
+			ldap_search_base=None,
+			ldap_query=None,
+			ldap_method=None,
+			auto_activate=False,
+			roles=None,
+			ldap_tls_reqcert=None,
+			ldap_bind_user=None,
+			ldap_bind_password = None
+		)
+
+	def on_settings_save(self, data):
+		old_flag = self._settings.get_boolean(["active"])
+		octoprint.plugin.SettingsPlugin.on_settings_save(self, data)
+		new_flag = self._settings.get_boolean(["active"])
+		if new_flag != old_flag:
+			if new_flag:
+				self._logger.warning("Warning! Activating LDAP Plugin")
+				settings().set(["accessControl", "userManager"],'octoprint_authldap.LDAPUserManager')
+				settings().save()
+			else:
+				if settings().get(["accessControl", "userManager"]) == 'octoprint_authldap.LDAPUserManager':
+					self._logger.warning("Deactivating LDAP Plugin")
+					settings().remove(["accessControl", "userManager"])
+					settings().save()
+
+	# TemplatePlugin
+
+	def get_template_configs(self):
+		return [
+			dict(type="settings", custom_bindings=False)
+		]
+
+
+__plugin_name__ = "Auth LDAP"
+
+
+def __plugin_load__():
+	global __plugin_implementation__
+	__plugin_implementation__ = LDAPUserManager()
+
+	global __plugin_hooks__
+	__plugin_hooks__ = {
+		"octoprint.plugin.softwareupdate.check_config":
```

Comment:
Removed the octoprint.users.factory hook and switched to the userManager setting.